dev/tools/test-migration.py

```
@@ -523,7 +593,21 @@ def validate_migration(active_pod_info):
         "upgrade-claim-warm missing storage-migrated-at annotation!"
+    assert claim_warm.get("status", {}).get("sandbox", {}).get("name") == "upgrade-pool-warm-a1b2c", \
+        f"upgrade-claim-warm lost its bound sandbox during upgrade! Expected upgrade-pool-warm-a1b2c, got: {claim_warm.get('status', {}).get('sandbox', {}).get('name')}"
     print("upgrade-claim-warm validation PASSED.")
+
+    print("Validating all 10 flake stress-test claims preserved their bound sandboxes...")
+    for i in range(1, 11):
+        claim_name = f"upgrade-claim-warm-flake-{i}"
+        claim_obj = claim_by_name.get(claim_name)
+        assert claim_obj is not None, f"Claim {claim_name} missing from migrated claims list!"
+        expected_sb = f"upgrade-pool-warm-flake-{i}"
+        assert actual_sb == expected_sb, \
```

Ruff (0.15.20)

[error] Undefined name `actual_sb`

(F821)